rero_ils/modules/holdings/tasks.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2021 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Celery tasks for holdings records."""
+
+from __future__ import absolute_import, print_function
+
+from celery import shared_task
+from flask import current_app
+
+from .api import Holding, HoldingsSearch
+from ..items.api import ItemsSearch
+
+
+@shared_task(ignore_result=True)
+def delete_standard_holdings_having_no_items():
+    """Removes standard holdings records with no attached items."""
+    current_app.logger.debug(
+        "Starting delete_standard_holdings_having_no_items task ...")
+    msg = 'Nothing to do'
+    
+    holdings = HoldingsSearch() \
+        .filter('term', holdings_type='standard') \
+        .source('pid') \
+        .scan()
+    try:
+        for holding in holdings:
+            holdings_pid = holding.pid
+            results = ItemsSearch().filter(
+            'term', holding__pid=holdings_pid).source().count()
+            if not results:
+                record = Holding.get_record_by_pid(holdings_pid)
+                record.delete(force=False, dbcommit=True, delindex=True)
+                msg = f'Removing standard holdings pid: {holdings_pid}'
+                current_app.logger.info(msg)
+    except Exception as err:
+        pass
```

Comment:
You should benefits of the `item_count` ES key into `holding` index to execute a single query :
```python
es_query = HoldingsSearch() \
    .filter('term', holdings_type='standard') \       
    .filter('term', item_count=0) \
    .source('pid') \        
    .scan()
 for hit in es_query:
     holding = Holding.get_record_by_pid(hit.pid)
     record.delete(force=False, dbcommit=True, delindex=True)
     msg = f'Removing standard holdings pid: {holdings_pid}'
     current_app.logger.info(msg)
```